Fix NuGet package missing content issues

nr-ahemsath opened the pull request:
## Description

Fixes #1346 

PR #1321 broke the following NuGet packages that we deployed in release 10.5.0:
NewRelic.Agent
NewRelic.Agent.Api
NewRelicWindowsAzure

These packages' `.nuspec` files previously did not have a `<files>...</files>` section.  When we added one to include the new icon file, we did not realize that if you have a files section, it disables automatic inclusion of content - everything you want to be in the package has to be in the `<files>...</files>` section.  This PR corrects that.

The team swarmed on these changes, including building the packages locally and comparing their content to the previous good release (10.4.0) so the PR review process is primary going to be testing the built artifacts from this PR's CI to make sure they do what we want them to do (since we currently don't have automatic testing of these artifacts).

# Author Checklist
- [ ] ~Unit tests, Integration tests, and Unbounded tests completed~
- [ ] ~Performance testing completed with satisfactory results (if required)~
- [x] [Agent Changelog](/src/Agent/CHANGELOG.md) or [Lambda Agent changelog](/src/AwsLambda/CHANGELOG.md) updated 

# Reviewer Checklist
- [ ] Perform code review
- [ ] Pull request was adequately tested (new/existing tests, performance tests)
- [ ] Review Changelog

tippmar-nr commented on build/Packaging/NugetAgent/NewRelic.Agent.nuspec:
```diff
@@ -26,6 +26,8 @@
     </contentFiles>
   </metadata>
   <files>
+    <file src="**/newrelic/**/*" target="." />
```
Nit: Might as well make the slash  / backslash usage consistent everywhere...

nr-ahemsath replied:
Good catch, done.

nr-ahemsath commented:
Notes on how I tested these changes for correctness:

For `NewRelic.Agent` and `NewRelic.Agent.Api`:

1. I started by creating a standard template web app project (.NET 6) in Visual Studio (2022)
2. I grabbed the `deploy-artifacts.zip` file from this PR's CI build, which contains the built .nuget files
3. I added the .nuget files for `NewRelic.Agent` and `NewRelic.Agent.Api` from this PR's CI build to a local NuGet package source, using the command: `nuget add $PATH_TO_NUGET_FILE -source C:\LocalNugets
4. I added "C:\LocalNugets" as a source in Visual Studio
5. I added both `NewRelic.Agent` and `NewRelic.Agent.Api` packages from the local source to my test web app.
6. I created a local folder publish profile and published the web app and verified that the expected files from those packages were there.
7. I repeated steps 5-6 but using the 10.4.0 NuGet packages from nuget.org, and a different local folder to publish to.
8. I compared the two publish folders, using Powershell (this technique described here: https://devblogs.microsoft.com/scripting/easily-compare-two-folders-by-using-powershell/)

All of the files that were in the publish folder for the 10.4.0 were also present in the publish folder for this CI's packages (the files may be different in content, but the names and paths are the same).

For `NewRelicWindowsAzure`:

Since this package is only useful for deploying the agent in an Azure Cloud Services environment, I couldn't test with a local package source.  I resorted to comparing the contents of this NuGet file with the 10.4.0 build, and noting that the same files are there in both (with the exception that the new package has the icon file that we want it to).
